### PR TITLE
Bugfix: initializing all tensors and parameters of the `ConditionalRandomField` on the proper device

### PR DESCRIPTION
**Pull request by oroszgy**
Fixes https://github.com/allenai/allennlp/issues/2884

For details see: https://github.com/allenai/allennlp/pull/5335

### CODE REVIEW

File: allennlp_models/tagging/models/crf_tagger.py [hunk 1]
```diff
@@ -296,6 +296,12 @@ def decode_top_k_tags(top_k_tags):
 
         return output_dict
 
+    @overrides
+    def cuda(self, device=None):
+        super().cuda(device)
+        self.crf.cuda(device)
+        return self
```
Comment (epwalsh) [inline allennlp_models/tagging/models/crf_tagger.py:303]: Is this really necessary? `self.crf` is a `nn.Module`, so I would think this already works fine without overriding this method.

**Comment by dirkgr** [comment]
What's the deal with this PR? The title makes it seem important.

File: allennlp_models/tagging/models/crf_tagger.py (hunk 1 above)
Comment (oroszgy) [reply allennlp_models/tagging/models/crf_tagger.py:303]: @epwalsh you were true, just checked it. We can safely remove this change. The bug is only in the `conditional_random_field.py`

**Comment by oroszgy** [comment]
@dirkgr I think you can close this PR, this should solve alone the problem: https://github.com/allenai/allennlp/pull/5335

File: allennlp_models/tagging/models/crf_tagger.py (hunk 1 above)
Comment (oroszgy) [reply allennlp_models/tagging/models/crf_tagger.py]: Fixed: https://github.com/allenai/allennlp/pull/5335/commits/84d4e20282ef05a16762cf322683e5b3512db8ab

**Pull request closed without merging**